### PR TITLE
Use the new re-entrant do-nothing script

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Just a draft so we can trial it in the upcoming release.

https://github.com/SciTools-incubator/nothing

- [x] need to set appropriate permissions for the new repo

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py:755]: 76beb65a4

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: a010bd3d9

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 0c9681058

File: tools/release_do_nothing.py
Comment (trexfeathers) [inline tools/release_do_nothing.py:517]: This doesn't actually work - the `bit_length()` of the string can be anywhere from `253` to `256`. The correct two things to check are:

- That the string can be validly parsed as a 16-bit number (`int(sha256_string, 16)`)
- That the string is 64 characters long

File: tools/release_do_nothing.py
Comment (HGWright) [inline tools/release_do_nothing.py:544]: This requires a `"conda deactivate"` to work

File: tools/release_do_nothing.py
Comment (trexfeathers) [inline tools/release_do_nothing.py:679]:
```suggestion
            "https://dev.azure.com/conda-forge/feedstock-builds/_build?definitionId=464"
```

File: tools/release_do_nothing.py
Comment (trexfeathers) [inline tools/release_do_nothing.py:753]:
```suggestion
            f"{self.git_tag} has been released."
```

File: tools/release_do_nothing.py
Comment (trexfeathers) [inline tools/release_do_nothing.py:625]: The changes are best made within the same PR that proposes the new release.

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: bc7f6bca2

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py:547]: 0f8721b47

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 78d0255c5

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 94fa686b5

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 25812b0c5

**Comment by pp-mo** [comment]
Tested this in use + it was handy
But it would be nice if the "nothing" package allowed you to restart from latest stored. 
I created a couple of issues to suggest improvements in the common code package : https://github.com/SciTools-incubator/nothing/issues/1 https://github.com/SciTools-incubator/nothing/issues/2

**Comment by trexfeathers** [comment]
> Tested this in use + it was handy But it would be nice if the "nothing" package allowed you to restart from latest stored. I created a couple of issues to suggest improvements in the common code package : [SciTools-incubator/nothing#1](https://github.com/SciTools-incubator/nothing/issues/1) [SciTools-incubator/nothing#2](https://github.com/SciTools-incubator/nothing/issues/2)

Thanks @pp-mo these are great suggestions! I don't consider them a blocker to this PR, which I know @HGWright is looking at today.

File: tools/release_do_nothing.py
Comment (ESadek-MO) [inline tools/release_do_nothing.py:25]:
```suggestion
    raise ImportError(install_message)
```

File: tools/release_do_nothing.py
Comment (ESadek-MO) [inline tools/release_do_nothing.py:187]:
```suggestion
                f'wget "{url}" -O {file};\n'
                f"git add {file};\n"
                "git commit -m 'Update CF standard names table.';\n"
                f"git push -u origin {working_branch};"
```

File: tools/release_do_nothing.py
Comment (trexfeathers) [inline tools/release_do_nothing.py:623]: Need to include a note to check that conda-forge automation has updated the feedstock `README` to reference the release candidate channel - this is considered evidence that things have worked correctly.

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py:707]:
```suggestion
```

File: tools/release_do_nothing.py
Comment (ESadek-MO) [inline tools/release_do_nothing.py:521]:
```suggestion
            valid = valid and len(sha256_string) == 64
```
I'm still figuring out what `&` does exactly, I think it does bitwise comparisons, so it's comparing `1` (`TRUE`) to `1000000` (`64`).

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
I think you're pretty much there. There are a few comments sitting on this PR, and I've added one of my own. I think once they're all done we merge this is, and if there's further issues we fix those when they arise.

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 4cf07a99bedfa8ab0ffc7b79ee1df17b48db9f3d

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py:689]: 8577b7914d251593bde0473e4d09ce493393120a

File: tools/release_do_nothing.py
Comment (trexfeathers) [reply tools/release_do_nothing.py]: 3e6fd1f99f6e5188711ef28e9f2a7510258ec374

I have tested this

**Review by ESadek-MO — APPROVED** [review]

**Comment by trexfeathers** [comment]
Thanks @ESadek-MO!

**Pull request merged**
